pyquil/reference_simulator.py

```diff
@@ -178,7 +178,16 @@ def sample_bitstrings(self, n_samples):
         if self.rs is None:
             raise ValueError("You have tried to perform a stochastic operation without setting the "
                              "random state of the simulator. Might I suggest using a PyQVM object?")
-        probabilities = np.real_if_close(np.diagonal(self.density))
+
+        # for np.real_if_close the actual tolerance is (machine_eps *tol_factor). If we use
+        tol_factor = 1e8
```

Comment:
Make `tol_factor` an optional input to the function, with default value `1e8` instead of declaring it mid-way in the function.

Reply:
I thought about this optional input too.

In the PR blurb above I mentioned

> Future work: It would be nice to have an optional argument that allowed the user to set the tolerance when truncating imaginary and negative probabilities. Exposing this feature will require some careful thought.

Given that we both thought of doing it must be a X idea where $X \in  \{good, bad\}$ ;)

Reply:
- [ ] todo